pageserver/src/import_datadir.rs

```diff
@@ -403,3 +425,219 @@ fn import_wal<R: Repository>(
+
+                // Extract the checkpoint record and import it separately.
+                let pg_control = ControlFileData::decode(&bytes[..])?;
+                let checkpoint_bytes = pg_control.checkPointCopy.encode()?;
+                modification.put_checkpoint(checkpoint_bytes)?;
+                info!("imported control file");
+
+                // Import it as ControlFile
+                modification.put_control_file(bytes)?;
+                return Ok(Some(pg_control));
+            }
+            "pg_filenode.map" => {
+                let bytes = read_all_bytes(reader)?;
+                modification.put_relmap_file(spcnode, dbnode, bytes)?;
+                info!("imported relmap file")
+            }
+            "PG_VERSION" => {
+                info!("ignored");
+            }
+            _ => {
+                import_rel(modification, file_path, spcnode, dbnode, reader, len)?;
+                info!("imported rel creation");
+            }
+        }
+    } else if file_path.starts_with("base") {
+        let spcnode = pg_constants::DEFAULTTABLESPACE_OID;
+        let dbnode: u32 = file_path
+            .iter()
+            .nth(1)
+            .unwrap()
+            .to_string_lossy()
+            .parse()
+            .unwrap();
+
+        match file_path.file_name().unwrap().to_string_lossy().as_ref() {
+            "pg_filenode.map" => {
+                let bytes = read_all_bytes(reader)?;
+                modification.put_relmap_file(spcnode, dbnode, bytes)?;
+                info!("imported relmap file")
+            }
+            "PG_VERSION" => {
+                info!("ignored");
+            }
+            _ => {
+                import_rel(modification, file_path, spcnode, dbnode, reader, len)?;
+                info!("imported rel creation");
+            }
+        }
+    } else if file_path.starts_with("pg_xact") {
+        let slru = SlruKind::Clog;
+
+        import_slru(modification, slru, file_path, reader, len)?;
+        info!("imported clog slru");
+    } else if file_path.starts_with("pg_multixact/offsets") {
+        let slru = SlruKind::MultiXactOffsets;
+
+        import_slru(modification, slru, file_path, reader, len)?;
+        info!("imported multixact offsets slru");
+    } else if file_path.starts_with("pg_multixact/members") {
+        let slru = SlruKind::MultiXactMembers;
+
+        import_slru(modification, slru, file_path, reader, len)?;
+        info!("imported multixact members slru");
+    } else if file_path.starts_with("pg_twophase") {
+        let xid = u32::from_str_radix(&file_path.file_name().unwrap().to_string_lossy(), 16)?;
+
+        let bytes = read_all_bytes(reader)?;
+        modification.put_twophase_file(xid, Bytes::copy_from_slice(&bytes[..]))?;
+        info!("imported twophase file");
+    } else if file_path.starts_with("pg_wal") {
+        panic!("found wal file in base section");
+    } else {
+        info!("ignored");
+    }
+
+    // TODO: pg_tblspc ??
```

Comment:
@hlinnaka no clue what's in `pg_tblspc`, but I copied this comment from `import_timeline_from_postgres_datadir`. Let me know if I should look into it now.

Reply:
Postgres tablespaces allow user to define datadir where to store the files.
see: https://www.postgresql.org/docs/current/manage-ag-tablespaces.html

It doesn't make sense in Neon "bottomless" setup, as we store everything in pageserver anyway.
We do not allow creating new tablespaces in neon, but imported user database may contain them...

Let's cover it with a test and error out here, until we fix it elsewhere.
I'll send a PR.

Reply:
`pg_basebackup` packs tablespaces into separate tars, so we will need to adjust API to support them.
Let's leave them as TODO, because neon clusters never use tablespaces.